src/iidm/converter/xml/DanglingLineXml.cpp

```diff
@@ -72,9 +76,18 @@ DanglingLine& DanglingLineXml::readRootElementAttributes(DanglingLineAdder& adde
 
 void DanglingLineXml::readSubElements(DanglingLine& dl, NetworkXmlReaderContext& context) const {
     context.getReader().readUntilEndElement(DANGLING_LINE, [this, &dl, &context]() {
-        if (context.getReader().getLocalName() == CURRENT_LIMITS) {
-            CurrentLimitsAdder<std::nullptr_t, DanglingLine> adder = dl.newCurrentLimits();
-            readCurrentLimits(adder, context.getReader());
+        if (context.getReader().getLocalName() == ACTIVE_POWER_LIMITS) {
+            IidmXmlUtil::assertMinimumVersion(getRootElementName(), ACTIVE_POWER_LIMITS, ErrorMessage::NOT_SUPPORTED, IidmXmlVersion::V1_5(), context);
+            IidmXmlUtil::runFromMinimumVersion(IidmXmlVersion::V1_5(), context.getVersion(), [&context, &dl]() {
+                readActivePowerLimits([&dl]() { return dl.newActivePowerLimits(); }, context.getReader());
```

Comment:
To be check: do we really need to create a lambda? Could we pass directly `d1.newActivePowerLimits` ?

I will also check if you really need a supplier here: why just passing the instance of the adder is not OK

Reply:
We can pass the adder as argument, but we have to pass it by rvalue (to avoid creating a temporary adder). Writing 
```
                readActivePowerLimits(dl.newActivePowerLimits(), context.getReader());
```
is only possible with writing
```
void AbstractConnectableXml<Added, Adder, Parent>::readActivePowerLimits(ActivePowerLimitsAdder&& activePowerLimitsOwner, const powsybl::xml::XmlStreamReader& reader, const stdcxx::optional<int>& index) {
```

Writing a non const reference lead to the following error
![image](https://user-images.githubusercontent.com/60604989/121199289-1a1f8680-c873-11eb-82c8-3f954563e22b.png)